Initialize Collections lazily, use List instead of LinkedList

fkleedorfer opened the pull request:
Reduces the memory footprint and speeds up object creation.

fkleedorfer committed:
Initialize Collections lazily, use List instead of LinkedList

Reduces the memory footprint and speeds up object creation.

fkleedorfer committed:
Initialize TypeVO's Collections lazily, improve static collections

Reduces the memory footprint and speeds up object creation.

quasarchimaere commented on src/main/java/com/buildingsmart/tech/ifcowl/vo/IFCVO.java:
```diff
@@ -18,18 +18,15 @@
     private String name = null;
-    private LinkedList<Object> list = new LinkedList<>();
-    private Map<String, LinkedList<IFCVO>> inversePointerSets = new HashMap<>();
+    private List<Object> list = null;
```
unnecessary null init

fkleedorfer replied:
fixed

fkleedorfer committed:
Initialize EntityVO's Collections lazily, improve static collections

Reduces the memory footprint and speeds up object creation.

fkleedorfer committed:
Remove null initializers for members

fkleedorfer committed:
Made constructors public again

It seems that client like to circumvent the internal caches, so these are needed.
